v2/yarpcgrpcfx/yarpcgrpcfx.go

```diff
@@ -0,0 +1,259 @@
+	ic := InboundConfig{}
+	if err := p.Provider.Get(_inboundConfigurationKey).Populate(&ic); err != nil {
+		return InboundConfigResult{}, err
+	}
+	return InboundConfigResult{
+		Config: ic,
+	}, nil
+}
+
+// StartInboundsParams defines the dependencies of this module.
+type StartInboundsParams struct {
+	fx.In
+
+	Lifecycle fx.Lifecycle
+	Router    yarpc.Router
+	Config    InboundConfig
+	Logger    *zap.Logger        `optional:"true"`
+	Tracer    opentracing.Tracer `optional:"true"`
+}
+
+// StartInbounds constructs and starts inbounds.
+func StartInbounds(p StartInboundsParams) error {
+	inbound := yarpcgrpc.Inbound{
+		Addr:   p.Config.Address,
+		Router: p.Router,
+		Tracer: p.Tracer,
+		Logger: p.Logger,
+	}
+	p.Lifecycle.Append(fx.Hook{
+		OnStart: func(ctx context.Context) error {
+			return inbound.Start(ctx)
+		},
+		OnStop: func(ctx context.Context) error {
+			return inbound.Stop(ctx)
+		},
+	})
+	return nil
+}
+
+// OutboundsConfig is the configuration for constructing a set of outbounds.
+type OutboundsConfig struct {
+	Outbounds map[string]OutboundConfig `yaml:",inline"`
+}
+
+// OutboundConfig is the configuration for constructing a specific outbound.
+type OutboundConfig struct {
+	// Specifies the outbound's service name.
+	//
+	// If not set, defaults to the configured outbound key.
+	Service string `yaml:"name"`
+
+	// Specifies the address to use for this Outbound.
+	Address string `yaml:"address"`
+
+	// Specifies the peer list chooser to use for this Outbound.
+	//
+	// If set, an address does not need to be configured.
+	Chooser string `yaml:"chooser"`
+}
+
+// OutboundsConfigParams defines the dependencies of this module.
+type OutboundsConfigParams struct {
+	fx.In
+
+	Provider config.Provider
+}
+
+// OutboundsConfigResult defines the values produced by this module.
+type OutboundsConfigResult struct {
+	fx.Out
+
+	Config OutboundsConfig
+}
+
+// NewOutboundsConfig produces an OutboundsConfig.
+func NewOutboundsConfig(p OutboundsConfigParams) (OutboundsConfigResult, error) {
+	oc := OutboundsConfig{}
+	if err := p.Provider.Get(_outboundConfigurationKey).Populate(&oc); err != nil {
+		return OutboundsConfigResult{}, err
+	}
+	return OutboundsConfigResult{
+		Config: oc,
+	}, nil
+}
+
+// ClientParams defines the dependencies of this module.
+type ClientParams struct {
+	fx.In
+
+	Lifecycle       fx.Lifecycle
+	Config          OutboundsConfig
+	Dialer          *yarpcgrpc.Dialer
+	ChooserProvider yarpc.ChooserProvider
+
+	Logger *zap.Logger        `optional:"true"`
+	Tracer opentracing.Tracer `optional:"true"`
+}
+
+// ClientResult defines the values produced by this module.
+type ClientResult struct {
+	fx.Out
+
+	Clients []yarpc.Client `group:"yarpcfx"`
+}
+
+// NewClients produces yarpc.Clients.
+func NewClients(p ClientParams) (ClientResult, error) {
+	var clients []yarpc.Client
+	for name, o := range p.Config.Outbounds {
+		var (
+			chooser yarpc.Chooser
+			addr    *url.URL
```

Comment:
nit: Let's update `yarpchttpfx` to use `addr` for the variable if we prefer it :)

Reply:
this was to avoid squashing the url package name :/